src/components/Header/utils.ts

```diff
@@ -0,0 +1,265 @@
+          }
+        }
+
+        // Process nested items if they exist
+        if (item.items && item.items.length > 0) {
+          navItem.items = item.items
+            .map((nestedItem) => {
+              if (!nestedItem) return null
+
+              const nestedNavItem: NavItem = {
+                id: nestedItem.id || generateRandomId(),
+              }
+
+              if (nestedItem.link) {
+                const convertedNestedLink = convertToNavLink(nestedItem.link)
+                if (convertedNestedLink) {
+                  nestedNavItem.link = convertedNestedLink
+                }
+              }
+
+              return nestedNavItem.link ? nestedNavItem : null
+            })
+            .filter(Boolean) as NavItem[]
+
+          // Only include items array if it has elements
+          if (navItem.items && navItem.items.length === 0) {
+            delete navItem.items
+          }
+        }
+
+        return navItem.link || (navItem.items && navItem.items.length > 0) ? navItem : null
+      })
+      .filter(Boolean) as NavItem[]
+
+    // Only include items if they exist
+    if (result.items && result.items.length === 0) {
+      delete result.items
+    }
+  }
+
+  // Only return the result if it has a link or items
+  return result.link || (result.items && result.items.length > 0) ? [result] : []
+}
+
+/**
+ * Helper function to convert a payload navLink (@/fields/navLinks) to a frontend NavLink
+ * Returns undefined if the link is invalid
+ */
+function convertToNavLink(link: {
+  type?: ('internal' | 'external') | null
+  reference?:
+    | ({
+        relationTo: 'pages'
+        value: number | Page
+      } | null)
+    | ({
+        relationTo: 'posts'
+        value: number | Post
+      } | null)
+  url?: string | null
+  label?: string | null
+  newTab?: boolean | null
+}): NavLink | undefined {
+  // Extract label from link or reference
+  let linkLabel: string | undefined = link.label || undefined
+
+  // Handle external links
+  if (link.type === 'external' && link.url) {
+    if (!linkLabel) {
+      return undefined // External links must have a label
+    }
+
+    return {
+      type: 'external',
+      label: linkLabel,
+      url: link.url,
+      newTab: !!link.newTab,
+    }
+  }
+
+  // Handle internal links with references
+  if (
+    link.type === 'internal' &&
+    link.reference &&
+    (link.reference.relationTo === 'pages' || link.reference.relationTo === 'posts')
+  ) {
+    const reference = link.reference
+
+    // If reference.value is a number, don't render this nav item
+    if (typeof reference.value === 'number') {
+      return undefined
+    }
+
+    // Try to get title from reference value if it exists
+    if (
+      !linkLabel &&
+      typeof reference.value === 'object' &&
+      reference.value &&
+      'title' in reference.value
+    ) {
+      linkLabel = (reference.value as any).title
+    }
+
+    // If we still don't have a label, don't render this nav item
```

Comment:
IIRC `title` is required on pages and posts, so we should `throw` here and hard-crash during CI or whatever, it would mean we somehow invalidated fundamental assumptions